plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+
+
+async def reload_mods():
+    """
+    Reload all moderators from data.
+    """
+    global _ticketmods
+
+    logger.debug("Loading ticket moderators.")
+    [mod.unload() for mod in _ticketmods.values()]
+    _ticketmods = {row[0]: TicketMod(row) for row in TicketMod._select_where()}
+    [await mod.load() for mod in _ticketmods.values()]
+    logger.info("Loaded {} ticket moderators.".format(len(_ticketmods)))
+
+
+def get_or_create_mod(modid) -> TicketMod:
+    """
+    Get a single TicketMod by modid, or create it if it doesn't exist.
+    """
+    mod = _ticketmods.get(modid, None)
+    if not mod:
+        mod = TicketMod(TicketMod._insert(modid=modid))
+        _ticketmods[modid] = mod
+    return mod
+
+
+# ------------ Commands ------------
+
+def resolve_ticket(msg, ticketarg=None) -> Ticket:
+    if ticketarg is not None and isinstance(ticketarg, commands.StringArg):
+        maybe_id = int(ticketarg.text)
+        if maybe_id < 2147483647:
+            tickets = fetch_tickets_where(id=maybe_id)
+        else:
+            tickets = fetch_tickets_where(list_msgid=maybe_id)
+        return next(tickets, None)
+    elif ref := msg.reference:
+        if (ref_msg := ref.resolved) and isinstance(ref_msg, discord.Message):
+            if ref_msg.author == discord_client.client.user and ref_msg.embeds:
+                embed = ref_msg.embeds[0]
+                if embed.author.name and embed.author.name.startswith("Ticket #"):
+                    ticket_id = int(embed.author.name[8:].split(' ', maxsplit=1)[0])
+                    tickets = fetch_tickets_where(id=ticket_id)
+                    return next(tickets, None)
+
+
+def summarise_tickets(*tickets, title="Tickets", fmt=None):
+    """
+    Create paged embeds of ticket summaries from the provided list of tickets.
+    """
+    if not tickets:
+        return None
+
+    lines = [ticket.summary(fmt=fmt) for ticket in tickets]
+    blocks = ['\n'.join(lines[i:i+10]) for i in range(0, len(lines), 10)]
+    page_count = len(blocks)
+
+    embeds = (discord.Embed(description=blocks[i], title=title) for i in range(page_count))
+
+    if page_count > 1:
+        embeds = (embed.set_footer(text="Page {}/{}".format(i+1, page_count)) for i, embed in enumerate(embeds))
+
+    return embeds
+
+
+Page = namedtuple('Page', ('content', 'embed'), defaults=(None, None))
+
+
+async def pager(dest: discord.abc.Messageable, *pages):
+    """
+    Page a sequence of pages.
+    """
+    _next_reaction = '⏭️'
+    _prev_reaction = '⏮️'
+
+    # Sanity check
+    if not pages:
+        raise ValueError("Cannot page with no pages!")
+
+    # Send first page
+    msg = await dest.send(**pages[0]._asdict())
+
+    if len(pages) == 1:
+        return
+
+    # Add reactions
+    await msg.add_reaction(_prev_reaction)
+    await msg.add_reaction(_next_reaction)
+
+    index = 0
+    while True:
+        try:
+            payload = await discord_client.client.wait_for(
```

Comment:
You may find `plugins.reactions.ReactionMonitor` useful

Reply:
I'll look into it